rust/protocol/src/sealed_sender.rs

```diff
@@ -1006,7 +1025,15 @@ pub async fn sealed_sender_multi_recipient_encrypt<R: Rng + CryptoRng>(
     Ok(serialized)
 }
 
-// For testing
+/// Split out the result of [sealed_sender_multi_recipient_encrypt] into a sequence of individual
+/// encrypted [UnidentifiedSenderMessageContent]s.
+///
+/// This method is primarily intended for testing in the context of the Signal app, but is also used
+/// e.g. in the jni bindings with
+/// [crate::protocol::SealedSender_MultiRecipientMessageForSingleRecipient] to allow processing a v2
+/// sealed-sender message directly by [sealed_sender_decrypt_to_usmc]. For the Signal app, this
+/// process of splitting out a v2 sealed-sender message into individual messages is typically
+/// performed implicitly by the Signal server.
 pub fn sealed_sender_multi_recipient_fan_out(data: &[u8]) -> Result<Vec<Vec<u8>>> {
```

Comment:
Ah, *both* uses are for unit testing (not app-level testing) – it's just that one is testing at the Rust level and the other is testing at the Java/Swift/TypeScript level. There's never a real-life need to encrypt something and immediately decrypt it, but that's what the tests do to simulate both sides of a connection. And this isn't the API the server would actually use, because the server needs to keep the association between messages and recipients. So it really is just for testing.

Reply:
Ah ok thank you! I think I was vaguely approaching this understanding since I posted this--it was at first somewhat confusing to me why this method was tossing away all the info we'd carefully added in `multi_recipient_encrypt()`!